### ISSUE TITLE
Crash in "oci_core_volume_attachments" datasource for listing Volume Attachments

### ISSUE THREAD

**Issue by rrcastro** [oracle/terraform-provider-oci#477]
### Terraform Version

<details>
Terraform v0.11.3 or v0.10.8
</details>

### OCI Provider Version

<details>
2018/03/13 14:09:52 [INFO] terraform-provider-oci 0.0.0

</details>
<p>2.1.0 - 2018-03-08: <a href="https://github.com/oracle/terraform-provider-oci/releases/download/v2.1.0/linux_amd64.tar.gz"> linux_amd64.tar.gz  </a></p></p>

### Description:
I am trying to list the Volume Attachments resources provisioned in a compartment. 
The below datasources.tf causes a crash of Terraform. I have tried it repeatedly and it is consistently crashing. 

<p><strong>datasources.tf</strong></p>
<pre><code>data "oci_core_volume_attachments" "test_volume_attachments" {
        compartment_id = "${var.compartment_ocid}"
        limit = 10
}
</code></pre>

### Terraform Plan

<details>
Terraform plan are crashing, so cannot provide the plan file. 

</details>

<p><p>crash.log is attached: <a href="https://github.com/oracle/terraform-provider-oci/files/1807130/crash.log">crash.log</a></p></p>

**Comment by rohanbhattacharjee** [comment]
I took a quick look ... it seems from the logs, that the following exception is being thrown by the provider code - https://github.com/oracle/terraform-provider-oci/blob/master/provider/core_volume_attachments_data_source.go#L135-L138

```
    iscsiVolumeAttachment, castOk := r.(oci_core.IScsiVolumeAttachment)
    if !castOk {
        panic("unexpected VolumeAttachment type on item.")
    }
```

What this code tells me is that at least one of the volume attachments in your compartment is not using `iscsi`.

Is there any way you can check if that's the case ?

**Comment by rrcastro** [comment]
Through the Web Console, all volume attachments in this compartiment are "iscsi".

**Comment by michaelhart** [comment]
This issue should be resolved with release v2.1.1:
https://github.com/oracle/terraform-provider-oci/releases/tag/v2.1.1

If you are still experiencing any issues after updating to the latest version, please reopen this ticket and let us know.

Thanks,
Michael
